tutorials/hardhat-tutorial.md

```diff
@@ -0,0 +1,425 @@
+
+The first fixture in the test file represents an "empty" stakingDAO but the second fixture will have one member (besides the DAO admin). Fixtures allow you to quickly switch between different states, simplifying and speeding up the testing process. In this tutorial only two fixtures are specified but it's easy to imagine other states you'd like to test, such as a scenario with 100 DAO members or one with multiple admins of the DAO. 
+
+To define the two fixtures, add the following snippet to your test file:
+
+``` javascript
+// `describe` is a Mocha function that allows you to organize your tests. All Mocha
+// functions are available in the global scope.
+//
+// `describe` receives the name of a section of your test suite, and a
+// callback. The callback must define the tests of that section. This callback
+// can't be an async function.
+describe("Dao contract", function () {
+  async function deployDaoFixture() {
+
+    // Get the ContractFactory and Signers here.
+    const [deployer] = await ethers.getSigners();
+    const delegationDao = await ethers.getContractFactory("DelegationDAO");
+    
+    // To deploy our StakingDao, you need to call delegationDao.deploy() and await
+    // its deployed() method, which happens once its transaction has been mined.
+    const deployedDao = await delegationDao.deploy(targetCollator, deployer.address);
+
+    await deployedDao.deployed();
+
+    // Fixtures can return anything you consider useful for your tests
+    return { deployedDao };
+  }
+
+    async function deployDaoFixtureWithMembers() {
+    const [deployer, member1] = await ethers.getSigners();
+
+    const delegationDao = await ethers.getContractFactory("DelegationDAO");
+    const deployedDao = await delegationDao.deploy(targetCollator, deployer.address);
+
+    await deployedDao.deployed();
+    await deployedDao.grant_member(member1.address);
+
+    return { deployedDao};
+  }
+```
+
+### Writing your First Test Cases {: #writing-your-first-test-cases }
+
+This tutorial deviates from the [test driven development philosophy](https://en.wikipedia.org/wiki/Test-driven_development){target=_blank} of writing your tests prior to your code. A few test cases will be cherry picked for demonstration purposes. 
+
+First, you'll create a subsection called `deployment` to keep the test file organized. This isn't required but may be helpful for organization purposes. Next you'll define your first test case by using the `it` mocha function. A description that clearly indicates what the test is doing is provided as well as indication that the function is async. This first test is simply checking to see that the StakingDAO is correctly storing the address of the target collator.
+
+Next, a fixture is loaded - in this case, the "empty" stakingDAO fixture. Fixtures are easily interchangeable within your test cases. 
+
```

Comment:
I would explicitly state that this snippet needs to be added to the describe dao contract function